fendermint/eth/api/src/state.rs

```diff
@@ -221,19 +224,23 @@ where
 
     /// Return the height of a block which we should send with a query,
     /// or None if it's the latest, to let the node figure it out.
+    ///
+    /// Adjusts the height of the query to +1 so the effects of the block is visible.
     pub async fn query_height(&self, block_id: et::BlockId) -> JsonRpcResult<FvmQueryHeight> {
         match block_id {
             et::BlockId::Number(bn) => match bn {
-                et::BlockNumber::Number(height) => Ok(FvmQueryHeight::from(height.as_u64())),
+                // The client might be asking by height of a block, expecting to see the results.
+                et::BlockNumber::Number(height) => Ok(FvmQueryHeight::from(height.as_u64() + 1)),
```

Comment:
Can we document how +1 here does not break overall API consistency? We might want to document this at the top as general comments.

Reply:
Thanks for keeping me honest, I'll add more comments. 

My thinking is that as long as we try to make sure that if someone asks for the latest block number we give them the last committed one -1, then _that_ block should be executed, and therefore it's safe to ask for +1 here. And if someone got a block from the subscription, that should be executed too, so again it's okay to send the query to +1, where the results are stored.